politico/api/v1/party/routes.py

```diff
@@ -0,0 +1,139 @@
+from flask import Blueprint, request, make_response, jsonify
+from politico.api.v1.party.modal import partyTable
+
+party = Blueprint('party', __name__)
+
+party_table = partyTable()
+
+@party.route('/parties', methods = ['GET'])
+def get_parties():
+    # route for displaying all parties
+    return make_response(jsonify({
+        'status': 200, 
+        'data': party_table.get_all_parties()
+    }), 200)
+
+@party.route('/parties/<int:id>', methods=['GET'])
+def get_one_party(id):
+    # route for getting a single party
+    party = party_table.get_single_party(id)
+    if not party:
+        return make_response(jsonify({
+            'status': 404,
+            'error': 'No party with id:{} found'.format(id)
+        }), 404)
+    else:
+        return make_response(jsonify({
+            'status': 200, 
+            'data': [party.party_data]
+        }), 200)
+
+@party.route('/parties', methods=['POST'])
+def add_party():
+    # route for adding a party
+    party_data = request.get_json()
+    msg = validateKeysInParty(party_data)
+    if msg != 'ok':
+        return make_response(jsonify({
+            'status': 406,
+            'error': msg
+        }), 406)
+    
+    msg1 = validateValueInParty(party_data)
+    if msg1 != 'ok':
+        return make_response(jsonify({
+            'status': 406,
+            'error': msg1
+        }), 406)
+
+    party_exists = party_table.get_single_party_by_name(party_data['name'])
+    if not party_exists:
+        added_party = party_table.add_party(party_data)
+        return make_response(jsonify({
+            'status': 201,
+            'data': [added_party]
+        }), 201)
+    else:
+        return make_response(jsonify({
+            'status': 409, 
+            'error': 'Party with name::{} already exists'.format(party_data['name'])
+        }), 409)
+
+@party.route('/parties/<int:id>', methods=['PATCH'])
+def update_party(id):
+    # route for updating a party
+    party_data = request.get_json()
+    msg = validateKeysInParty(party_data)
+    if msg != 'ok':
+        return make_response(jsonify({
+            'status': 406,
```

Comment:
I'd discourage the use of 406 status code here as it has more to do with headers. Use 400 instead which means Bad Request

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406